api/v1alpha1/envoyproxy_types.go

```diff
@@ -86,8 +86,46 @@ type EnvoyProxySpec struct {
 	//
 	// +optional
 	Shutdown *ShutdownConfig `json:"shutdown,omitempty"`
+
+	// FilterOrder defines the order of filters in the Envoy proxy's HTTP filter chain.
+	//
+	// +optional
+	FilterOrder []FilterOrder `json:"filterOrder,omitempty"`
```

Comment:
we are technically assigning priority - `0` , `100` etc, should this be `priority` instead of `order` ?

Reply:
I'm open to both. My choice may not be the most accurate as a non-native speaker.

Reply:
Thought again, `priority` may be more appropriate because the number defined here is not exactly the `order` of the filter in the chain.